### PR TITLE
Python program for Carmicheal Number

### PR DESCRIPTION
**Pull request by shubhajitroy123**
### Describe your change:
Added Python Program for checking whether a number is Carmicheal number or not.

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
maths/carmichael_number.py:3:89: E501 line too long (104 > 88 characters)
maths/carmichael_number.py:48:89: E501 line too long (89 > 88 characters)

File: maths/carmichael_number.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+"""
+== Carmichael Numbers ==
+A number n is said to be a Carmichael number if it satisfies the following modular arithmetic condition:
```
Comment (cclauss) [inline maths/carmichael_number.py:3]: Too long.  88 chars max.

File: maths/carmichael_number.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                return False
+        b = b + 1
+    return True
+
+
+if __name__ == "__main__":
+    print("Program to check whether a number is a Carmichael Numbers or not...")
+    number = int(input("Enter number: ").strip())
+    print(
+        f"{number} is {'' if isCarmichaelNumber(number) else 'not '}a Carmichael Number."
```
Comment (cclauss) [inline maths/carmichael_number.py:48]: Too long.  88 chars max.

File: maths/carmichael_number.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    power(b, n-1) MOD n = 1,
+    for all b ranging from 1 to n such that b and
+    n are relatively prime, i.e, gcd(b, n) = 1
+
+Examples of Carmichael Numbers: 561, 1105, ...
+https://en.wikipedia.org/wiki/Carmichael_number
+"""
+
+
+def gcd(a, b):
```
Comment (cclauss) [inline maths/carmichael_number.py:14]: You can use https://docs.python.org/3/library/math.html#math.gcd if you want.

```suggestion
def gcd(a: int, b: int) -> int:
    """
    >>> gcd(4, 36)
    4
    >>> gcd(4, 38)
    2
    """
```
These are [doctests](https://docs.python.org/3/library/doctest.html) which are **_required_** for all functions.

**Comment by shubhajitroy123** [comment]
Hey @cclauss, can you please review this pull request for Hacktoberfest.

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please write doctests for your functions

File: maths/carmichael_number.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+def isCarmichaelNumber(n: int) -> bool:
+    b = 2
+    while b < n:
+
+        if gcd(b, n) == 1:
+
+            if power(b, n - 1, n) != 1:
+                return False
+        b = b + 1
```
Comment (cclauss) [inline maths/carmichael_number.py:41]:
```suggestion
        if gcd(b, n) == 1 and power(b, n - 1, n) != 1:
            return False
        b += 1
```
All these single-letter character variable names make the code look like Fortran from the 1970's.

**Review by cclauss — APPROVED** [review]

**Comment by CaedenPH** [comment]
@cclauss 
Why have you merged this commit? It didn't even pass the `CONTRIBUTING` guidelines such as `snake_case` naming conventions and informative parameter names

**Comment by cclauss** [comment]
Yes.  I saw that as I pressed the Merge button.  If you open a PR to fix that then I will merge it.

**Comment by cclauss** [comment]
In fact, we should probably add a good flake8 naming plugin to pre-commit so that our tests fail on bad names.
* https://github.com/DmytroLitvinov/awesome-flake8-extensions#naming

Your thoughts?  https://github.com/DmytroLitvinov/awesome-flake8-extensions#naming

**Comment by CaedenPH** [comment]
> In fact, we should probably add a good flake8 naming plugin to pre-commit so that our tests fail on bad names.
> 
> * https://github.com/DmytroLitvinov/awesome-flake8-extensions#naming
> 
> Your thoughts?

Sounds like a good idea

**Comment by CaedenPH** [comment]
> Do you have `pre-commit` running on your local machine?
> 
> Would you be willing to make the following one-line change to `.pre-commit-config.yaml` and then fix the files that will be flagged by running `pre-commit run --all-files`? It is a bit of busy-work that will take some time but it will be a nice improvement to our project.
> 
> ```
> diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
> index 0abe647..2f6a928 100644
> --- a/.pre-commit-config.yaml
> +++ b/.pre-commit-config.yaml
> @@ -40,6 +40,7 @@ repos:
>            - --ignore=E203,W503
>            - --max-complexity=25
>            - --max-line-length=88
> +        additional_dependencies: [pep8-naming]
> 
>    - repo: https://github.com/pre-commit/mirrors-mypy
>      rev: v0.982
> ```

Unfortunately I am not at home currently, but I can do it in a couple hours when I return

**Comment by cclauss** [comment]
No rush!

**Pull request merged**